massa-pos-exports/src/pos_final_state.rs

```diff
@@ -711,6 +711,10 @@ impl PoSFinalState {
     pub fn get_all_roll_counts(&self, cycle: u64) -> BTreeMap<Address, u64> {
         let db = self.db.read();
 
+        if self.get_cycle_index(cycle).is_none() {
+            panic!("Cycle {} not in history", cycle)
```

Comment:
Is It mandatory to kill the node ?

Reply:
Not really mandatory, but we should never query this info on a cycle that is not on disk. If it's the case, I would not trust the state our node is in, so we should panic to be safe and avoid unexpected consequences of that IMO.